Sources/Paywalls/Events/PaywallEventStore.swift

```diff
@@ -85,23 +85,61 @@ internal actor PaywallEventStore: PaywallEventStoreType {
 @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
 extension PaywallEventStore {
 
-    static func createDefault(documentsDirectory: URL?) throws -> PaywallEventStore {
+    static func createDefault(
+        applicationSupportDirectory: URL?,
+        documentsDirectory: URL? = nil
+    ) throws -> PaywallEventStore {
+        let url = Self.url(in: try applicationSupportDirectory ?? Self.applicationSupportDirectory)
+        Logger.verbose(PaywallEventStoreStrings.initializing(url))
+
         let documentsDirectory = try documentsDirectory ?? Self.documentsDirectory
-        let url = documentsDirectory
+        Self.removeStoreIfExists(Self.url(in: documentsDirectory))
```

Comment:
maybe we can call this something like `removeLegacyDirectoryIfExists` so it's obvious at first glance what we're doing and why?